### ISSUE TITLE
"discard queue event"  should still allow event to be used

### ISSUE THREAD

**Issue by TApplencourt** [intel/llvm#12388]
**Is your feature request related to a problem? Please describe**

Discard event ( [sycl_ext_oneapi_discard_queue_events](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/supported/sycl_ext_oneapi_discard_queue_events.asciidoc) is really useful.

Sadly, it restricts any usage of the event.  Meaning that, I cannot create a queue with this property and then pass it to oneMKL for example.

Pseudo code
```  sycl::property_list props{ext::oneapi::property::queue::discard_events{},
                            property::queue::in_order{}};
  sycl::queue Q( props );
  oneMKL::dgmem(Q, ...);  // This will fail, as dgem use event internally
  Q.wait();

**Describe the solution you would like**

It's easy. Just implement `discard_event` like [HipSYCL is doing it](https://github.com/AdaptiveCpp/AdaptiveCpp/blob/develop/doc/extensions.md#hipsycl_ext_coarse_grained_events). Meaning when using the `event` (either for `.wait` or `depend_on`) just submit a `Q.wait()`. 

This will solve the composability issue. 

**Additional context**

See more info here: https://github.com/KhronosGroup/SYCL-Docs/issues/404).

**Comment by Pennycook** [comment]
@0x12CC : I'm not sure.  Such a change would permit something that previously wasn't allowed, so I don't think it would count as a "breaking change".  But I'm reluctant to change `sycl_ext_oneapi_discard_queue_events` at this stage, primarily because I don't think people should be writing new code that uses it.  [`sycl_ext_oneapi_enqueue_functions`](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_oneapi_enqueue_functions.asciidoc) is our preferred solution to enqueuing commands without events.  @gmlueck, do you agree?

**Comment by gmlueck** [comment]
Yes, I was thinking that we would probably deprecate and then remove "sycl_ext_oneapi_discard_queue_events" now that we have "[sycl_ext_oneapi_enqueue_functions](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_oneapi_enqueue_functions.asciidoc)"

**Comment by TApplencourt** [comment]
The barrier of entry is not the same. For big applications, `sycl_ext_oneapi_discard_queue_events` is a one-line change, porting all their submissions to `sycl_ext_oneapi_enqueue_functions` is a monthly effort.

**Comment by gmlueck** [comment]
> The barrier of entry is not the same. For big applications, `sycl_ext_oneapi_discard_queue_events` is a one-line change, porting all their submissions to `sycl_ext_oneapi_enqueue_functions` is a monthly effort.

Is it really that big of a change?  Porting to `sycl_ext_oneapi_enqueue_functions` is still a localized change in the code, affecting just the few lines that submit a kernel.  Th ere is no need to change the entire application, either.  You can just change one kernel submission if you want, leaving the remaining submissions to use the old API.

**Comment by TApplencourt** [comment]
True, It's just that people have more kernels (100+) than queues. 
The implementation cost seems low to me, so it is a high reward, but that makes sense if you prefer to wait for the "official" fix.
@abagusetty, what do you think? You where the one looking for this :)

**Comment by abagusetty** [comment]
@TApplencourt That should work as well, Thanks will give  it a try.
